src/MediatR/IMediatorContext.cs

```diff
@@ -0,0 +1,21 @@
+using System.Collections.Generic;
+using System.Threading;
+
+namespace MediatR
+{
+    /// <summary>
+    /// Context
+    /// </summary>
+    public interface IMediatorContext
+    {
+        /// <summary>
+        /// Cancellation Token
+        /// </summary>
+        CancellationToken CancellationToken { get; set; }
+
+        /// <summary>
+        ///  Gets or sets a key/value collection that can be used to share data within the scope of this request/notification.
+        /// </summary>
+        IDictionary<object, object> Items { get; set; }
```

Comment:
Make sure whatever you use as a key in this dictionary has an implementation for Equals() and GetHashCode() and that it's not mutable or you're going to have a bad day.

https://www.knightlycode.com/2017/06/21/back-to-the-basics-custom-types-in-c-dictionary-key/
https://blog.markvincze.com/back-to-basics-dictionary-part-3-built-in-gethashcode/